src/tester.zig

```diff
@@ -0,0 +1,184 @@
+// Copyright 2024 TerseTS Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Provides methods for testing TerseTS.
+
+const std = @import("std");
+const ArrayList = std.ArrayList;
+const rand = std.rand;
+const Random = rand.Random;
+const Allocator = std.mem.Allocator;
+const Error = std.mem.Allocator.Error;
+const math = std.math;
+const time = std.time;
+const testing = std.testing;
+const debug = std.debug;
+
+const tersets = @import("tersets.zig");
+const Method = tersets.Method;
+
+/// Number of values to generate for testing.
+const number_of_values = 50;
+
+/// Test that values generated by `uncompressed_values_generator` are within `error_bound` according
+/// to `within_error_bound` after they have been compressed and decompressed using `method`. The
+/// libraries public interface is used to make it simpler refactor the libraries internals.
+pub fn testGenerateCompressAndDecompress(
+    uncompressed_values_generator: fn (
```

Comment:
Are we sure that `uncompressed_values_generator` should be snake_case? Since it is callable I think it should be camelCase! https://ziglang.org/documentation/0.12.0/#Style-Guide

Reply:
Good catch, this should know be fixed for all higher-order functions.